### PR TITLE
`pub` our config + chainspec types and fields (in the node)

### PR DESCRIPTION
**Pull request by dwerner**
We need to `pub` all the fields and parsing of `chainspec.toml`, `config.toml`, and `accounts.toml` for anyone who wants to read/write config files that the node can understand. While not officially "public", they are still effectively already and implicitly part of our public api, and therefore a source of endless pitfalls for anyone trying to generate this information or configure things programmatically. AND we already have this code.

Where we had getters, but no setters, the getters have been removed and replaced with inline field accessors.

While this is a large changeset, it should -not- change any behavior. We are merely exposing through the `casper-node` crate types and fields that were previously not pub.

### CODE REVIEW

**Review by georgepisaltu — APPROVED** [review]
LGTM

**Review by marc-casperlabs — APPROVED** [review]
I'm all for making public just-a-bunch-of-data type's fields. I don't think there's any internal invariants we require?

File: node/src/reactor/main_reactor/config.rs [hunk 1]
```diff
@@ -12,22 +12,39 @@ use crate::{
...
-    pub(crate) diagnostics_port: DiagnosticsPortConfig,
-    pub(crate) block_accumulator: BlockAccumulatorConfig,
-    pub(crate) block_synchronizer: BlockSynchronizerConfig,
-    pub(crate) upgrade_watcher: UpgradeWatcherConfig,
+pub struct Config {
+    /// Config values for the node.
+    pub node: NodeConfig,
+    /// Config values for logging.
+    pub logging: LoggingConfig,
+    /// Config values for logging.
```
Comment (rafal-ch) [inline node/src/reactor/main_reactor/config.rs]:
```suggestion
    /// Config values for consensus.
```

File: node/src/reactor/main_reactor/config.rs [hunk 2]
```diff
@@ -12,22 +12,39 @@ use crate::{
...
+    pub consensus: ConsensusConfig,
+    /// Config values for logging.
```
Comment (rafal-ch) [inline node/src/reactor/main_reactor/config.rs]:
```suggestion
    /// Config values for network.
```

File: node/src/reactor/main_reactor/config.rs [hunk 3]
```diff
@@ -12,22 +12,39 @@ use crate::{
...
+    pub network: NetworkConfig,
+    /// Config values for network.
```
Comment (rafal-ch) [inline node/src/reactor/main_reactor/config.rs:22]:
```suggestion
    /// Config values for event stream server.
```

File: node/src/types/chainspec.rs [hunk 4]
```diff
@@ -56,25 +54,38 @@ pub const CHAINSPEC_FILENAME: &str = "chainspec.toml";
...
     #[serde(rename = "core")]
-    pub(crate) core_config: CoreConfig,
+    pub core_config: CoreConfig,
+
+    /// Highway config.
     #[serde(rename = "highway")]
-    pub(crate) highway_config: HighwayConfig,
+    pub highway_config: HighwayConfig,
+
+    /// Deploy COnfig
```
Comment (rafal-ch) [inline node/src/types/chainspec.rs]:
```suggestion
    /// Deploy Config
```

File: node/src/types/chainspec/accounts_config.rs [hunk 5]
```diff
@@ -43,21 +44,25 @@ pub struct AccountsConfig {
 }
 
 impl AccountsConfig {
+    /// Creates a new `AccountsConfig.
```
Comment (rafal-ch) [inline node/src/types/chainspec/accounts_config.rs]:
```suggestion
    /// Creates a new `AccountsConfig`.
```

**Review by rafal-ch — APPROVED** [review]
LGTM :+1: 
Approving with few minor nits (mostly typos).

**Comment by dwerner** [comment]
bors merge

**Pull request merged**